source/distributed/solution_transfer.cc

```diff
@@ -264,7 +264,7 @@ namespace parallel
                   // on all children from the associated FECollection.
                   std::set<unsigned int> fe_indices_children;
                   for (unsigned int child_index = 0;
-                       child_index < GeometryInfo<dim>::max_children_per_cell;
+                       child_index < cell->n_children();
```

Comment:
This is basically some clean-up on the fly. I could move it into a different PR if you want to.